test/AxelarGateway.js

```diff
@@ -1372,6 +1763,91 @@ describe('AxelarGateway', () => {
 
             expect(values).to.deep.eq([name, symbol, decimals, cap, amount1, amount2]);
         });
+
+        it('should skip unknown command', async () => {
+            const data = buildCommandBatch(
+                await getChainId(),
+                [getRandomID(), getRandomID(), getRandomID(), getRandomID()],
+                ['deployToken', 'unknownCommand', 'mintToken', 'mintToken'],
+                [
+                    getDeployCommand(name, symbol, decimals, cap, ethers.constants.AddressZero, 0),
+                    '0x',
+                    getMintCommand(symbol, owner.address, amount1),
+                    getMintCommand(symbol, wallets[1].address, amount2),
+                ],
+            );
+
+            const input = await getSignedWeightedExecuteInput(
+                data,
+                operators,
+                getWeights(operators),
+                threshold,
+                operators.slice(0, threshold),
+            );
+
+            await expect(gateway.execute(input, getGasOptions())).to.emit(gateway, 'TokenDeployed');
+
+            const tokenAddress = await gateway.tokenAddresses(symbol);
+
+            expect(tokenAddress).to.be.properAddress;
+
+            const token = burnableMintableCappedERC20Factory.attach(tokenAddress);
+
+            const values = await Promise.all([
+                token.name(),
+                token.symbol(),
+                token.decimals(),
+                token.cap().then(bigNumberToNumber),
+                token.balanceOf(owner.address).then(bigNumberToNumber),
+                token.balanceOf(wallets[1].address).then(bigNumberToNumber),
+            ]);
+
+            expect(values).to.deep.eq([name, symbol, decimals, cap, amount1, amount2]);
+        });
+
+        it('should not execute the same commandID twice', async () => {
+            const duplicateCommandID = getRandomID();
+            const data = buildCommandBatch(
+                await getChainId(),
+                [getRandomID(), getRandomID(), duplicateCommandID, duplicateCommandID],
+                ['deployToken', 'mintToken', 'mintToken', 'transferOperatorship'],
```

Comment:
I don't see why the other commands are needed here for testing duplicate checks. Also, let's avoid using expensive commands like token deployments unless the test is focused on that. I recommend using approveContractCall instead since it's the cheapest and doesn't require a token to be setup for the test. This could simplify other tests too.

And feel free to create helper functions where applicable to reduce code duplication

Reply:
I replaced all the commands with approveContractCall to save gas (since all other commands are tested separately anyways), I also combined two of the tests into one because I realized there was a coverage overlap. These changes allowed me to turn the beforeEach to a before for deploying the gateway. Ran the batch commands on fantom and they were significantly faster.